types/index.d.ts

```diff
@@ -322,6 +331,8 @@ export interface ChannelListProps {
     e: Client.Event,
   ): void;
   setActiveChannelOnMount?: boolean;
+  /** Optional function to filter channels prior to loading in the DOM */
+  filterFunction?: <T>(channels: T) => T;
```

Comment:
can we think of a better name for this function? `filterFunction is not very intuitive`